chapter_02_repository.asciidoc

```diff
@@ -422,11 +422,37 @@ find them again. Our in memory data would let us add new objects, just like a
 list or a set, and since the objects are in memory we never need to call a
 "Save" method, we just fetch the object we care about, and modify it in memory.
 
-The ideal repository has just two methods: `add` to put a new item in the
+The simplest repository has just two methods: `add` to put a new item in the
 repository, and `get` to return a previously added item. We stick rigidly to
 using these methods for data access in our domain and our _service layer_. This
 self-imposed simplicity stops us from coupling our domain model to the database.
-//TODO what about delete and update?
+
+[[abstract_repo]]
+.The simplest possible repository (repository.py)
+====
+[source,python]
+----
+
+class AbstractRepository(abc.ABC):
+
+    @abc.abstractmethod
+    def add(self, batch):
+        raise NotImplementedError
+
+    @abc.abstractmethod
+    def get(self, reference):
+        raise NotImplementedError
+----
+====
+
+//TODO: what about .list()?
```

Comment:
The *ideal* repository has add/get. 

Why do you need to list aggregates if you're only modifying one at a time?
Delete should usually be handled as a soft-delete, ie. `batch.cancel`.
Update is taken care of by the unit of work and change detection.

If a repository is just an infinitely big in-memory store for our objects, then the only thing you need is add/get. In practice, repository interfaces often get cluttered with `finder` methods, but strictly speaking those could either be on some separate interface, or treated as read-side concerns.